### PR TITLE
Bring up building/deployment process to current standard

### PR DESCRIPTION
**Pull request by JayH5**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,11 +36,11 @@ The `/app` directory is created and set as the current working directory. This i
 Running a built-in worker class without a config file:
 ```shell
 docker run --rm -it \
-  -e WORKER_CLASS=vumi.blinkenlights.MetricTimeBucket \
   -e AMQP_HOST=rabbitmq.service.consul \
   -e VUMI_OPT_BUCKETS=3 \
   -e VUMI_OPT_BUCKET_SIZE=10 \
-  praekeltfoundation/vumi
+  praekeltfoundation/vumi \
+    --worker-class vumi.blinkenlights.MetricTimeBucket
```
Comment (JayH5) [inline README.md]: This is a change to the example to show mixing environment variables and command line options. The old example still works.

File: vumi-entrypoint.sh [hunk 2]
```diff
@@ -1,39 +1,62 @@
...
-fi
-
-SENTRY_OPT="${SENTRY_DSN:+--sentry $SENTRY_DSN}"
-
-SET_OPTS=$(env \
-  | grep ^VUMI_OPT_ \
-  | sed -e 's/^VUMI_OPT_//' -e 's/=/ /' \
-  | awk '{printf("%s=%s:%s ", "--set-option", tolower($1), $2);}')
+# If `python` isn't available, we should use `pypy`
+python=$(command -v python > /dev/null && echo python || echo pypy)
```
Comment (JayH5) [inline vumi-entrypoint.sh]: This is necessary since praekeltfoundation/docker-py-base#62

File: vumi-entrypoint.sh [hunk 3]
```diff
@@ -1,39 +1,62 @@
...
-    "$@"
+"$python" -c 'from twisted.plugin import IPlugin, getPlugins; list(getPlugins(IPlugin))'
+
+is_twistd_command() {
+	local cmd="$1"; shift
+	"$python" - <<EOF
+import sys
+from twisted.plugin import getPlugins
+from twisted.application.service import IServiceMaker
+sys.exit(0 if '$cmd' in [p.tapname for p in getPlugins(IServiceMaker)] else 1)
```
Comment (JayH5) [inline vumi-entrypoint.sh]: Twisted is complicated... but I found this here: https://github.com/twisted/twisted/blob/twisted-17.5.0/src/twisted/application/twist/_options.py#L174-L182

File: .travis.yml [hunk 4]
```diff
@@ -1,36 +1,38 @@
...
 before_deploy:
+  - pip install docker-ci-deploy==0.3.0
   - docker login -u "$REGISTRY_USER" -p "$REGISTRY_PASS"
 deploy:
   provider: script
-  script: ./deploy.sh "$IMAGE_TAG" ${TAG_DEFAULT:+--default}
+  script: dcd --version "$version" --version-latest "$image_tag"
   on:
-    branch: develop
+    branch: master
```
Comment (JayH5) [inline .travis.yml]: I'm slowly moving most of these `docker-` repos to use `master` as the default branch. The production images come from this branch and `develop` is not the best name for that.

**Comment by smn** [comment]
👍 thanks @JayH5 !

**Pull request merged**